File: cs2dart_roslyn_worker/Models/SymbolTable.cs
namespace Cs2DartRoslynWorker.Models;

/// <summary>
/// A dictionary mapping node IDs to resolved symbols.
/// Every named-reference node in a NormalizedSyntaxTree has a corresponding entry.
/// </summary>
public sealed class SymbolTable
{
    /// <summary>
    /// Maps node identity (assigned during normalization) to its resolved symbol.
    /// </summary>
    public Dictionary<int, ResolvedSymbol> Entries { get; set; } = new();

    /// <summary>Returns the ResolvedSymbol for nodeId, or null if not present.</summary>
    public ResolvedSymbol? Lookup(int nodeId) =>
        Entries.TryGetValue(nodeId, out var symbol) ? symbol : null;
}

File: lib/src/roslyn_frontend/models/symbol_table.dart
import 'resolved_symbol.dart';

/// A dictionary mapping node IDs to resolved symbols.
///
/// Every named-reference node in a [NormalizedSyntaxTree] has a corresponding
/// entry — either a fully-resolved [ResolvedSymbol] or one with
/// [SymbolKind.unresolved] as the sentinel.
final class SymbolTable {
  /// Maps node identity (assigned during normalization) to its resolved symbol.
  final Map<int, ResolvedSymbol> entries;

  const SymbolTable({required this.entries});

  /// Returns the [ResolvedSymbol] for [nodeId], or null if not present.
  ResolvedSymbol? lookup(int nodeId) => entries[nodeId];
}
